src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -289,6 +294,66 @@ public void processArguments() {
         if (!cli.isBlank() && !loaded.isEmpty()) {
             uiCommands.add(new UiCommand.OpenDatabases(loaded));
         }
+
+        if (cli.isCheckConsistency()) {
+            checkConsistency(cliPreferences, entryTypesManager);
+        }
+    }
+
+    private void checkConsistency(CliPreferences cliPreferences,
+                                  BibEntryTypesManager entryTypesManager) {
+        String fileName = cli.getCheckConsistency();
+        Optional<String> outputFormat = Optional.ofNullable(cli.getCheckConsistencyOutputFormat());
+
+        if (fileName == null) {
+            System.out.println(Localization.lang("No file specified for consistency check."));
+            return;
+        }
+
+        try {
+            Path filePath = Path.of(fileName);
+            ParserResult pr = OpenDatabase.loadDatabase(filePath, cliPreferences.getImportFormatPreferences(), fileUpdateMonitor);
+            BibDatabaseContext databaseContext = pr.getDatabaseContext();
+            List<BibEntry> entries = databaseContext.getDatabase().getEntries();
+
+            BibliographyConsistencyCheck consistencyCheck = new BibliographyConsistencyCheck();
+            BibliographyConsistencyCheck.Result result = consistencyCheck.check(entries);
+
+            if (outputFormat.isEmpty()) {
+                writeFindings(result);
+                return;
+            }
```

Comment:
Just assume txt as output  format - and write it to System.out - Should be possible with `new OutputStreamWriter(System.out)`.

Reply:
Comment still relevant.